umbraco/Umbraco.CMS.Backoffice
# Feature: (user) Collection Pagination

Merge after this PR: https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/925

This PR adds pagination to the collection context.

It includes a Pagination Manager to help with the pagination calculations. It will return values for the Total Number of Pages based on a page size and total number of collection items + calculate the skip value sent to the server.

Currently, the PaginationManager is used as part of the CollectionContext but can be used in any place that needs pagination.

**Test:**
The pagination is currently implemented in the user collection. You can test the functionality there. To ease testing, you can lower the page size on the user collection in this file: 
https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/feature/user-collection-pagination/src/packages/user/user/collection/user-collection.context.ts#L9 Then you don't have to create that many users.

* Make sure you can navigate between the pages.
* Navigate to the first and last pages.
* Filter the users based on state and user group (don't worry about styling of the drop-downs). Make sure pagination behaves correctly. 
* Filter users from the search bar. Make sure pagination behaves correctly.

https://github.com/umbraco/Umbraco.CMS.Backoffice/assets/6078361/f1632623-314a-4a97-beac-b76dc813c8e1
